src/development/ui/advanced/shaders.md

```diff
@@ -0,0 +1,228 @@
+---
+title: Writing and using Fragment Shaders
+description: How to author and use fragment shaders.
+short-title: Fragment Shaders
+---
+
+Note: The HTML backend does not support the FragmentProgram API. The CanvasKit
+backend will support this API in a future release.
+
+A shader is a program that is executed by the GPU, usually authored in a small,
+dart-like language such as GLSL. User authored shaders can be incorporated into
+Flutter applications and packages via the FragmentProgram API. These can be
+used to provide rich graphical effects beyond those provided by the Flutter SDK.
+
+## Adding Shaders to an Application
+
+Shaders must be declared in the new `shaders` section of your application or
+package pubspec.yaml file. This allows the Flutter command line tool to compile
+the shader to the appropriate backend format, as well as generate the necessary
+runtime metadata. When running in debug mode, changes to a shader program will
+trigger a recompilation and update of the shader during hot reload or hot
+restart.
+
+```yaml
+flutter:
+  shaders:
+    - shaders/myshader.frag
+```
+
+Shaders from packages can be included using the same technique as other assets.
+
+#### Loading shaders at runtime
+
+FragmentPrograms can be loaded at runtime using the [FragmentProgram.fromAsset]
+constructor. The key should be the file path specified in the pubspec.yaml
+above.
+
+```dart
+void loadMyShader() async {
+  var program = await FragmentProgram.fromAsset('shaders/myshader.frag');
+}
+```
+
+The FragmentProgram object can be used to create one or more FragmentShader
+instances. A FragmentShader represents a particular combination of a
+program and bound uniforms.
+
+```dart
+void updateShader(Canvas canvas, Rect rect, FragmentProgram program) {
+  var shader = program.createShader();
+  shader.setFloat(0, 42.0);
+  canvas.drawRect(rect, Paint()..shader = shader);
+}
+```
+
+## Canvas API
+
+Fragment shaders can be used with most Canvas APIs that accept either solid
+colors or gradients via a [Paint] object. And will cover an area determined
```

Comment:
Check the first two sentences here. Not sure what they're saying.

Reply:
I think what I'm trying to say is:

You can use shaders with most canvas APIs.
The shader works in the same way that a solid color or gradient works when applied to a shape
It doesn't work with certain APIs, like how a LinearGradient is ignored if set on the paint passed to drawImage

Reply:
Did you make this change, @jonahwilliams? The suggested text is clearer.